app/src/lib/editors/win32.ts

```diff
@@ -16,15 +16,32 @@ interface IWindowsAppInformation {
   installLocation: string
 }
 
-type ExpectedInstallationChecker = (
-  displayName: string,
-  publisher: string
-) => boolean
-
 type RegistryKey = { key: HKEY; subKey: string }
 
+type WindowsExternalEditorPathInfo =
+  | {
+      /**
+       * Registry key with the install location of the app. If not provided,
+       * 'InstallLocation' will be used.
+       **/
+      readonly installLocationRegistryKey?: 'InstallLocation'
+
+      /**
+       * List of lists of path components from the editor's installation folder to
+       * the potential executable shims. Only needed when the install location
+       * registry key is `InstallLocation`.
+       **/
+      readonly executableShimPaths: ReadonlyArray<ReadonlyArray<string>>
+    }
+  | {
+      /**
+       * Registry key with the install location of the app.
+       **/
+      readonly installLocationRegistryKey: 'DisplayIcon'
+    }
```

Comment:
This addresses the issue mentioned in https://github.com/desktop/desktop/pull/12863#pullrequestreview-741609918

Maybe it's a bit overkill? But I think it's more honest about the problem at hand… 🤔